dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/search/query/SearchForTypesTest.java

```diff
@@ -0,0 +1,44 @@
+package edu.harvard.iq.dataverse.search.query;
+
+import org.junit.jupiter.api.Test;
+
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.junit.Assert.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class SearchForTypesTest {
+
+    // -------------------- TESTS --------------------
+
+    @Test
+    public void byType() {
+        // when
+        SearchForTypes typesToSearch = SearchForTypes.byTypes(SearchObjectType.DATAVERSES, SearchObjectType.FILES);
+        // then
+        assertThat(typesToSearch.getTypes(), containsInAnyOrder(SearchObjectType.DATAVERSES, SearchObjectType.FILES));
+    }
+
+    @Test
+    public void byType__multiple_same_types() {
+        // when
+        SearchForTypes typesToSearch = SearchForTypes.byTypes(SearchObjectType.DATAVERSES, SearchObjectType.DATAVERSES);
+        // then
+        assertThat(typesToSearch.getTypes(), containsInAnyOrder(SearchObjectType.DATAVERSES));
```

Comment:
If we are testing whether the result has only one occurrence of DATAVERSES element, then this assertion is insufficient.

Reply:
Actually it is. From javadoc we have `containsInAnyOrder()` checks for collection sizes. To be sure I changed it to `contains()`

Reply:
My fault then.